### PR TITLE
Fix/Clarify lock object usage

### PR DESCRIPTION
**Pull request by carpawell**
See https://github.com/nspcc-dev/neofs-node/issues/1227#issuecomment-1138913967 for details.

Think it could wait until the whole implementation is ready and testing to prevent any breaking changes in the future (in fact I think that PR is already a breaking change).

### CODE REVIEW

**Review by realloc — DISMISSED** [review]

File: lock/types.proto [hunk 1]
```diff
@@ -9,7 +9,8 @@ import "refs/types.proto";
 
 // Lock objects protects a list of objects from being deleted. The lifetime of a
 // lock object is limited similar to regular objects in
-// `__NEOFS__EXPIRATION_EPOCH` attribute.
+// `__NEOFS__EXPIRATION_EPOCH` attribute. Lock object MUST have expiration epoch.
+// It is impossible to deleted lock object via ObjectService.Delete RPC call.
```
Comment (fyrchik) [inline lock/types.proto:13]: s/deleted/delete/

**Review by fyrchik — APPROVED** [review]

File: lock/types.proto [hunk 2]
```diff
@@ -9,7 +9,8 @@ import "refs/types.proto";
 
 // Lock objects protects a list of objects from being deleted. The lifetime of a
 // lock object is limited similar to regular objects in
-// `__NEOFS__EXPIRATION_EPOCH` attribute.
+// `__NEOFS__EXPIRATION_EPOCH` attribute. Lock object MUST have expiration epoch.
+// It is impossible to delete lock object via ObjectService.Delete RPC call.
```
Comment (fyrchik) [inline lock/types.proto:13]: `to delete A lock object`, sorry, missed that before.

**Review by fyrchik — DISMISSED** [review]

**Review by realloc — APPROVED** [review]

**Comment by alexvanin** [comment]
> Think it could wait until the whole implementation is ready and testing to prevent any breaking changes in the future (in fact I think that PR is already a breaking change).

Decided not to wait for some tests?

**Comment by carpawell** [comment]
> Decided not to wait for some tests?

Well, seems like that.

**Pull request merged**